src/storage/remote/read.c

```diff
@@ -144,6 +144,7 @@ storageReadRemote(THIS_VOID, Buffer *buffer, bool block)
                         protocolClientDataEndGet(this->client);
                     }
 
+                    result += this->remaining;
```

Comment:
What about rewriting the code below to achieve something like this:
```
            // Read if not eof
            if (!this->eof)
            {
                // If the buffer can contain all remaining bytes
                size_t remains = min(this->remaining, bufRemains(buffer));
                bufCatSub(buffer, this->block, bufUsed(this->block) - this->remaining, remains);
                result += remains;
                this->remaining -= remains;
                if (remains >= this->remaining)
                {
                    bufFree(this->block);
                    this->block = NULL;
                }
            }
``` 
?

Reply:
done